scripts/west_commands/runners/jlink.py

```diff
@@ -153,20 +153,29 @@ def supports_nogui(self):
         # -nogui was introduced in J-Link Commander v6.80
         return version.parse(ver) >= version.parse("6.80")
 
+    def supports_thread_info(self):
+        # The RTOSPlugin_Zephyr was introduced in 7.11b
+        return version.parse(ver) >= version.parse("7.11")
```

Comment:
You need to read the version before you parse it.
```
$ west -vv debug
ZEPHYR_BASE=/home/maureen/zephyrproject/zephyr (origin: configfile)
-- west debug: rebuilding
cmake version 3.13.2 is OK; minimum version is 3.13.1
Running CMake: /home/maureen/.local/bin/cmake --build /home/maureen/zephyrproject/zephyr/build
ninja: no work to do.
-- west debug: using runner jlink
runner_config: RunnerConfig(build_dir='/home/maureen/zephyrproject/zephyr/build', board_dir='/home/maureen/zephyrproject/zephyr/boards/arm/mimxrt685_evk', elf_file='/home/maureen/zephyrproject/zephyr/build/zephyr/zephyr.elf', hex_file=None, bin_file='/home/maureen/zephyrproject/zephyr/build/zephyr/zephyr.bin', gdb='/home/maureen/.local/zephyr-sdk-0.12.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gdb', openocd='/home/maureen/.local/zephyr-sdk-0.12.2/sysroots/x86_64-pokysdk-linux/usr/bin/openocd', openocd_search='/home/maureen/.local/zephyr-sdk-0.12.2/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts')
Traceback (most recent call last):
  File "/home/maureen/.local/bin/west", line 11, in <module>
    sys.exit(main())
  File "/home/maureen/.local/lib/python3.6/site-packages/west/app/main.py", line 779, in main
    app.run(argv or sys.argv[1:])
  File "/home/maureen/.local/lib/python3.6/site-packages/west/app/main.py", line 106, in run
    self.run_command(argv)
  File "/home/maureen/.local/lib/python3.6/site-packages/west/app/main.py", line 338, in run_command
    self.run_extension(args.command, argv)
  File "/home/maureen/.local/lib/python3.6/site-packages/west/app/main.py", line 408, in run_extension
    command.run(args, unknown, self.topdir, manifest=self.manifest)
  File "/home/maureen/.local/lib/python3.6/site-packages/west/commands.py", line 129, in run
    self.do_run(args, unknown)
  File "/home/maureen/zephyrproject/zephyr/scripts/west_commands/debug.py", line 34, in do_run
    do_run_common(self, my_args, runner_args)
  File "/home/maureen/zephyrproject/zephyr/scripts/west_commands/run_common.py", line 228, in do_run_common
    runner.run(command_name)
  File "/home/maureen/zephyrproject/zephyr/scripts/west_commands/runners/core.py", line 491, in run
    self.do_run(command, **kwargs)
  File "/home/maureen/zephyrproject/zephyr/scripts/west_commands/runners/jlink.py", line 166, in do_run
    if self.supports_thread_info():
  File "/home/maureen/zephyrproject/zephyr/scripts/west_commands/runners/jlink.py", line 158, in supports_thread_info
    return version.parse(ver) >= version.parse("7.11")
NameError: name 'ver' is not defined
```

Reply:
Thanks, pushed an old version, now fixed.

Reply:
Also caching it now, to avoid checking multiple times.